src/proto/h2/mod.rs

```diff
@@ -47,16 +47,15 @@ fn strip_connection_headers(headers: &mut HeaderMap, is_request: bool) {
             warn!("TE headers not set to \"trailers\" are illegal in HTTP/2 requests");
             headers.remove(TE);
         }
-    } else {
-        if headers.remove(TE).is_some() {
-            warn!("TE headers illegal in HTTP/2 responses");
-        }
+    } else if headers.remove(TE).is_some() {
+        warn!("TE headers illegal in HTTP/2 responses");
     }
 
     if let Some(header) = headers.remove(CONNECTION) {
+        static STATIC_CONNECTION: HeaderName = CONNECTION;
         warn!(
             "Connection header illegal in HTTP/2: {}",
-            CONNECTION.as_str()
+            STATIC_CONNECTION.as_str()
```

Comment:
I don't understand this change. Why is this better?

Reply:
this is a subtle one. It's the result of [this clippy lint](https://rust-lang.github.io/rust-clippy/master/#declare_interior_mutable_const), which is a hard failure.

Reply:
i've moved this change to a separate pull request.